server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -571,6 +571,44 @@ public Snapshot newSnapshotFromGen(long minGeneration) throws IOException {
         }
     }
 
+    /**
+     * Reads and returns the operation from the given location if the generation it references is still available. Otherwise
+     * this method will return <code>null</code>.
+     */
+    public Operation readOperation(Location location) throws IOException {
+        BaseTranslogReader reader = null;
+        Closeable onClose = null;
+        try (ReleasableLock ignored = readLock.acquire()) {
+            ensureOpen();
+            if (location.generation < getMinFileGeneration()) {
+                return null;
+            }
+            if (current.generation == location.generation) {
+                // fsync here to ensure all buffers are written to disk
+                current.syncUpTo(location.translogLocation + location.size);
```

Comment:
why require this?

Reply:
Because the buffer that we have in memory need to be written to disk if we point to it from the record. I will add a comment